28.md

```diff
@@ -0,0 +1,128 @@
+
+NIP-28
+======
+
+Public Chat
+-----------
+
+`draft` `optional` `author:ChristopherDavid` `author:fiatjaf` `author:jb55`
+
+This NIP defines new event kinds for public chat channels, channel messages, and basic client-side moderation.
+
+It reserves five event kinds (40-44) for immediate use and five event kinds (45-49) for future use.
+
+- `40 - channel create`
+- `41 - channel metadata`
+- `42 - channel message`
+- `43 - hide message`
+- `44 - mute user`
+
+Client-centric moderation gives client developers discretion over what types of content they want included in their apps, while imposing no additional requirements on relays.
+
+## Kind 40: Create channel
+
+Create a public chat channel.
+
+In the channel creation `content` field, Client SHOULD include basic channel metadata (`name`, `about`, `picture` as specified in kind 41).
+
+```json
+{
+    "content": "{\"name\": \"Demo Channel\", \"about\": \"A test channel.\", \"picture\": \"https://placekitten.com/200/200\"}",
+    ...
+}
+```
+
+
+## Kind 41: Set channel metadata
+
+Update a channel's public metadata.
+
+Clients and relays SHOULD handle kind 41 events similar to kind 0 `metadata` events.
+
+Clients SHOULD ignore kind 41s from pubkeys other than the kind 40 pubkey.
+
+Clients SHOULD support basic metadata fields:
+
+- `name` - string - Channel name
+- `about` - string - Channel description
+- `picture` - string - URL of channel picture
+
+Clients MAY add additional metadata fields.
+
+```json
+{
+    "content": "{\"name\": \"Updated Demo Channel\", \"about\": \"Updating a test channel.\", \"picture\": \"https://placekitten.com/201/201\"}",
+    "tags": ["e", <channel_create_event_id>],
+    ...
+}
+```
+
+
+## Kind 42: Create channel message
+
+Send a text message to a channel.
+
+```json
+{
+    "content": <string>,
+    "tags": ["e", <channel_create_event_id>],
+    ...
+}
+```
+
+## Kind 43: Hide message
+
+User no longer wants to see a certain message.
+
+The `content` may optionally include metadata such as a `reason`.
+
+Clients SHOULD hide event 42s shown to a given user, if there is an event 43 from that user matching the event 42 `id`.
```

Comment:
this could work for kind1 messages as well, it's something I've wanted for hiding replies to posts in damus.

Reply:
This could be done by a special reaction, like '!' or something like that, which if emitted should signify that the comment replied to should not be shown to the given user. Its a sort of 'hide' indicator.